### DISCUSSION TITLE
Parameterized builds

### DISCUSSION THREAD

**Discussion by mikeborozdin** [codemagic-ci-cd/codemagic-docs#2101] [Ideas]
### Use case

```Text
* Specifying app version before running a build
* Specifying a new version change log
```

### Idea

```Text
* Parametereized builds
* Input parameters for a build in YAML
* That are entered in the UI when pressing 'Run Build'
```

**Comment by Ycoi** [comment]
Thanks for the idea @mikeborozdin! Do you have some specific use case in mind for this feature? I see that you want to specify the app version and change log. Is this for specifiying those for a release build or something else. And do you possibly have other use cases in mind?

**Reply by Ycoi** [reply]
Hi @mikeborozdin. Getting back to this. We are considering implementing something like this so any more information about your use cases would be helpful.

Alternatively, if you would be open to chatting about this in a call, just answer me here and we can arrange one. That way we can talk more about your needs and use cases in person.

**Comment by boat-jitrapon** [comment]
Hi maintainers,

I have a specific use case for this. Say that my repo needs a dependency from another repository to build (like a jar file). My workflow begins by checkout the other repo to build the jar file from a branch, then proceeds to build the current repo. I would like the ability to set a parameter, specifically the branch of the other repo. I could use the environment variable to have a default branch name, but also for a specific build, I would like to be able to override that default parameter. Something like this could be useful for every build. Other use case would be the ability to specify some variables that need to be passed in to a shell script.

**Comment by mikhail-tokarev** [comment]
sorry for the late update 🙈 
This feature has been added some time ago. Read more from our docs https://docs.codemagic.io/knowledge-codemagic/build-inputs/
